libs/sdk-ui-kit/src/Datepicker/utils/tests/overlay.test.ts

```diff
@@ -0,0 +1,357 @@
+// (C) 2020 GoodData Corporation
+import { getOptimalAlignment, getOptimalAlignmentForRegion } from "../overlay";
+
+function generateOverlay() {
+    const boundaryRegion = {
+        top: 100,
+        left: 100,
+        width: 1000,
+        height: 1000,
+    };
+
+    const targetRegion = {
+        top: 900,
+        left: 900,
+        width: 100,
+        height: 100,
+    };
+
+    const selfRegion = {
+        top: 0,
+        left: 0,
+        width: 200,
+        height: 200,
+    };
+
+    const alignPoints = [
+        {
+            align: "tr tl",
+        },
+        {
+            align: "tl tr",
+        },
+    ];
+
+    return {
+        boundaryRegion,
+        targetRegion,
+        selfRegion,
+        alignPoints,
+    };
+}
+
+function generateOverlayWithCustomAlign(align: string) {
+    const boundaryRegion = {
+        top: 0,
+        left: 0,
+        width: 1000,
+        height: 1000,
+    };
+
+    const targetRegion = {
+        top: 50,
+        left: 50,
+        width: 100,
+        height: 100,
+    };
+
+    const selfRegion = {
+        top: 0,
+        left: 0,
+        width: 200,
+        height: 200,
+    };
+
+    const alignPoints = [
+        {
+            align,
+        },
+    ];
+
+    return {
+        boundaryRegion,
+        targetRegion,
+        selfRegion,
+        alignPoints,
+    };
+}
+
+describe("Overlay utils", () => {
+    it("should align inside document when aligned element is outside of the viewport", () => {
+        const { alignPoints, targetRegion, selfRegion } = generateOverlay();
+
+        const getViewportRegion = () => ({
+            left: 0,
+            top: 0,
+            width: 500,
+            height: 500,
+        });
+
+        const getDocumentRegion = () => ({
+            left: 0,
+            top: 0,
+            width: 1000,
+            height: 2500,
+        });
+
+        const optimalAlignment = getOptimalAlignment({
+            targetRegion,
+            selfRegion,
+            alignPoints,
+            getViewportRegion,
+            getDocumentRegion,
+        });
+
+        expect(optimalAlignment).toEqual({
+            alignment: {
+                left: 700,
+                right: 100,
+                top: 900,
+                align: "tl tr",
+            },
+            visiblePart: 1,
+        });
+    });
```

Comment:
i suggest to replace toEqual(hardcoded_value) with toMatchSnapshot(). same assertion except that the computer maintains the hardcoded values instead of us.

Reply:
snapshoting used